packs/splunk/actions/search.py

```diff
@@ -20,7 +21,16 @@ def __init__(self, config):
             scheme=self.config.get('scheme'))
 
     def run(self, query):
-        kwargs_oneshot = {"output_mode": "json"}
-        result = self.service.jobs.oneshot(query, **kwargs_oneshot)
-
-        return result.read()
+        result = self.service.jobs.oneshot(query, params={"output_mode": "json"})
+        reader = results.ResultsReader(result)
+        search_results = []
+
+        for item in reader:
+            if isinstance(item, results.Message):
+                # Diagnostic messages may be returned in the results
+                print '%s: %s' % (item.type, item.message)
```

Comment:
Is it intentional that you are using print? You might want to use the logger from the action.